module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,58 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+void printMap(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    for (auto arr : bitmap) {
+        for (auto el : arr) {
+            if (iscntrl(el))
+                std::cout << " ";
+            else
+                std::cout << el;
+        }
+        std::cout << "\n";
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& grayScale) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(width * height);
+    uint8_t count = 0;
+
+    for (auto row : grayScale) {
+        auto _prev = row.front();
```

Comment:
Maybe I am wrong, but I have heard that names which starts from **underscore** are reserved by convention for standard libraries or kernel space, base e.g. [Wikipedia's source](https://en.wikipedia.org/wiki/Naming_convention_(programming)#C_and_C++)

Reply:
It's a good practice not to use an `_` prefix anyway.